src/Microsoft.AspNetCore.BuildTools.ApiCheck/Program.cs

```diff
@@ -59,53 +58,87 @@ public static int Main(string[] args)
                 });
 
                 return app.Execute(args);
+            }
+            catch (FileNotFoundException e)
+            {
+                Console.WriteLine($"{e.GetType().FullName}: {e.Message}, file '{e.FileName}'");
+
+                var inner = e.InnerException;
+                while (inner != null)
+                {
+                    Console.WriteLine($"  {inner.GetType().FullName}: {inner.Message}");
+                    inner = inner.InnerException;
+                }
+
+                if (e.InnerException != null)
+                {
+                    Console.WriteLine("  --- End of inner exceptions ---");
+                }
 
+                Console.WriteLine(e.StackTrace);
```

Comment:
> This whole block can be simplified to `Console.WriteLine(e.ToString())`.

Unfortunately `FileNotFoundException`s that do not mention the filename sometimes bubble here. I will check if `ReflectionTypeLoadException` overrides `ToString()` to include loader exceptions in its output.

> consider using `Console.Error.WriteLine`

Why? MSBuild doesn't use `Console.Error` at all.